optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/steps/UpgradeStep.java

```diff
@@ -27,16 +27,16 @@ public abstract class UpgradeStep {
   // failing conversion of test indices that don't have lookups
   @VisibleForTesting @Setter protected boolean skipIndexConversion = false;
 
-  protected UpgradeStep(IndexMappingCreator index) {
+  protected UpgradeStep(final IndexMappingCreator index) {
     this.index = index;
   }
 
   public abstract UpgradeStepType getType();
 
   protected abstract void performUpgradeStep(SchemaUpgradeClient<?, ?> schemaUpgradeClient);
 
-  public void execute(SchemaUpgradeClient<?, ?> schemaUpgradeClient) {
-    if (!skipIndexConversion) {
+  public void execute(final SchemaUpgradeClient<?, ?> schemaUpgradeClient) {
+    if (!skipIndexConversion && index != null) {
```

Comment:
this is the fix. There are two subclasses that do not set an index property because they relate to index/template deletion. This then fails on execution. For such cases, we do not want to try to convert an index because no index exists.

The other commit in this PR is unrelated, but uses the updated logback property since the version has been updated